src/ChildrenAwareNodeInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Dhii\Data\Tree;
+
+use Dhii\Data\Hierarchy\ChildrenAwareInterface;
+use Traversable;
+
+/**
+ * A node that can have its children retrieved.
+ *
+ * @since [*next-version*]
+ */
+interface ChildrenAwareNodeInterface extends
+    NodeInterface,
+    ChildrenAwareInterface
+{
+    /**
+     * Retrieves child nodes of this node.
+     *
+     * @since [*next-version*]
+     *
+     * @return BasicNodeInterface[]|Traversable A list of child nodes.
```

Comment:
`BasicNodeInterface` is no longer in this package. I recommend changing this to `NodeInterface`.